app/src/main/java/com/sun/mygif/data/source/remote/response/DataResponseHandler.kt

```diff
@@ -0,0 +1,41 @@
+package com.sun.mygif.data.source.remote.response
+
+import com.sun.mygif.data.model.DataRequest
+import com.sun.mygif.utils.FORMAT_EXCEPTION_IO_HTTP
+import com.sun.mygif.utils.METHOD_GET
+import com.sun.mygif.data.source.remote.build
+import com.sun.mygif.data.source.remote.getJsonString
+import org.json.JSONException
+import java.io.IOException
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+
+
+interface DataResponseHandler<T> {
+
+    @Throws(JSONException::class)
+    fun parseToObject(jsonData: String): T
+
+    @Throws(IOException::class, JSONException::class)
+    fun getResponse(request: DataRequest): T? {
```

Comment:
đầu vào của class này a suggest e nên để là 1 url
```suggestion
    fun getResponse(url: String): T? {
```